var/spack/repos/builtin/packages/slate/package.py

```diff
@@ -54,9 +57,14 @@ class Slate(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("blaspp ~cuda", when="~cuda")
     depends_on("blaspp +cuda", when="+cuda")
     depends_on("blaspp ~rocm", when="~rocm")
+    depends_on("lapackpp ~cuda", when="~cuda")
+    depends_on("lapackpp +cuda", when="+cuda")
+    depends_on("lapackpp ~rocm", when="~rocm")
```

Comment:
Ditto: `slate~cuda` doesn't necessarily imply `lapackpp~rocm` and `blaspp~rocm`. Again, an edge case that maybe isn't important.

Reply:
If someone really needs to do that and we want to support it, it could be done, but it would complicate the spack package and would make it more difficult for the users who just want to turn on or off cuda nad rocm support for all components (slate, lapackpp, and blaspp).

Reply:
One addition: this just replicates the previous behavior such that slate depends on lapackpp in the same way that slate has previously depended on blaspp.